### PR TITLE
extraction-4-create-2nd-transform-rename-jsx-component - CTR-20

### PR DESCRIPTION
**Pull request by kiprasmel**
how to review: https://github.com/pipedrive/CodeshiftCommunity/pull/9 (commit-by-commit is still best)

### CODE REVIEW

File: packages/reusable-transforms/src/rename-jsx-component/verifyIsJsxIdentifierOrAddComment.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    j: JSCodeshift, //
+    path: ASTPath<any>,
+    name: string,
+    node: any,
+): node is namedTypes.JSXIdentifier => {
+    if (j.JSXIdentifier.check(node)) {
+        return true;
+    }
+
+    const comment = toComment([
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/rename-jsx-component/verifyIsJsxIdentifierOrAddComment.ts:17]: Is the output single line?

File: packages/utils/src/comments.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ function clean(value: string) {
     .trim();
 }
 
+export const inlineCommentPrefix = `\n\nTODO CODEMOD` as const;
```
Comment (martikaljuve) [inline packages/utils/src/comments.ts:11]: I would prefer colon (`TODO:`) for possibly better IDE support.

File: packages/reusable-transforms/src/rename-jsx-component/verifyIsJsxIdentifierOrAddComment.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    const comment = joinWith(" ")([
+        `ERROR expected`, //
+        name,
+        `to be`,
+        tick(j.jsxIdentifier("").type, ","), //
+        `got`,
+        tick(node.type, "."), //
+    ]);
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/rename-jsx-component/verifyIsJsxIdentifierOrAddComment.ts:24]: I know we discussed this already, but I'll note it here as well that to me, a template literal would be more readable to me, compared to this array concatenation.

```suggestion
    const comment = `ERROR expected ${name} to be \`${j.jsxIdentifier("").type}\`, got \`${node.type}\`.`;
```

**Review by martikaljuve — APPROVED** [review]

File: packages/utils/src/comments.ts (hunk 2 above)
Comment (kiprasmel) [reply packages/utils/src/comments.ts]: updated.

File: packages/reusable-transforms/src/rename-jsx-component/verifyIsJsxIdentifierOrAddComment.ts (hunk 3 above)
Comment (kiprasmel) [reply packages/reusable-transforms/src/rename-jsx-component/verifyIsJsxIdentifierOrAddComment.ts]: done.

**Pull request merged**